elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDictionary.java

```diff
@@ -0,0 +1,126 @@
+     * <ol>
+     *     <li> {@link MetricAggregation}
+     *     <li> {@link MetricComputation}
+     * </ol>
+     *
+     * @param fieldName  The entity field
+     *
+     * @return {@code true} if the field is a metric field
+     */
+    public boolean isMetricField(Class<?> cls, String fieldName) {
+        return attributeOrRelationAnnotationExists(cls, fieldName, Metric.class);
+    }
+
+    public static boolean isAnalyticView(Class<?> cls) {
+        return cls.isAnnotationPresent(FromTable.class) || cls.isAnnotationPresent(FromSubquery.class);
+    }
+
+    /**
+     * Maps a logical entity attribute into a physical SQL column name.
+     *
+     * @param cls The entity class.
+     * @param fieldName The entity attribute.
+     * @return The physical SQL column name.
+     */
+    public String getColumnName(Class<?> cls, String fieldName) {
+        Column[] column = getAttributeOrRelationAnnotations(cls, Column.class, fieldName);
+
+        // this would only be valid for dimension columns
+        JoinColumn[] joinColumn = getAttributeOrRelationAnnotations(cls, JoinColumn.class, fieldName);
+
+        if (column == null || column.length == 0) {
+            if (joinColumn == null || joinColumn.length == 0) {
+                return fieldName;
+            } else {
+                return joinColumn[0].name();
+            }
+        } else {
+            return column[0].name();
+        }
+    }
+
+    /**
+     * Maps a logical entity path into a physical SQL column name.
+     *
+     * @param path entity join path
+     * @return joined column name
+     */
+    public String getJoinColumn(Path path) {
+        Path.PathElement last = path.lastElement().get();
+        Class<?> lastClass = last.getType();
+
+        return getColumnName(lastClass, last.getFieldName());
+    }
+
+    /**
+     * Get alias for an entity class.
+     *
+     * @param entityClass entity class
+     * @return alias
+     */
+    public static String getClassAlias(Class<?> entityClass) {
+        return FilterPredicate.getTypeAlias(entityClass);
+    }
+
+    /**
+     * Maps an entity class to a physical table of subselect query, if neither {@link javax.persistence.Table}
+     * nor {@link Subselect} annotation is present on this class, use the class alias as default.
+     *
+     * @param cls The entity class.
+     * @return The physical SQL table or subselect query.
+     */
+    public String getTableOrSubselect(Class<?> cls) {
+        Subselect subselectAnnotation = getAnnotation(cls, Subselect.class);
+
+        if (subselectAnnotation == null) {
+            javax.persistence.Table tableAnnotation =
+                    getAnnotation(cls, javax.persistence.Table.class);
+
+            return (tableAnnotation == null)
+                    ? getJsonAliasFor(cls)
+                    : tableAnnotation.name();
+        } else {
+            return "(" + subselectAnnotation.value() + ")";
```

Comment:
I don't think we should do parenthetic work here.  That is part of query building.  This function should just return the name (minus the parenthesis).